extractor.py

```diff
@@ -1,146 +1,179 @@
 
-    attrs = {
-            _tar_ext: _tar_attr,
-            _targz_ext: _targz_attr,
-            _gzip_ext: _gzip_attr,
-            _zip_ext: _zip_attr,
-            _7z_ext: _7z_attr,
-    }
+def _create_configuration_file():
+    data = {_configs_str: []}
 
-    def get_full_command(self, prog, ext, filename):
-        if not prog:
-            print("Could not define which program we should use to extract your file");
-            return None
-        if not ext:
-            print("Could not define the extension of your file");
-            return None
-        if not filename:
-            print("Could not get the name of your file")
-            return None
-        return prog + ' ' + self.attrs[ext] + ' ' + filename
+    with open(config.EXTRACTOR_CONFIG_FILE, 'w') as outfile:
+        json.dump(data, outfile, indent=2)
 
-    def get_proper_ext(self, extension):
-        if not extension:
-            return None
-        if len(extension) > 2:
-            print("We don't support this extension.")
-            return None
-        if extension[0] is None:
-            if extension[1] is None:
-                print("We don't support this extension.")
-                return None
-            return self.exts[extension[1]]
-        else:
-            if extension[1] is None:
-                return self.exts[extension[0]]
-            _ext0 = self.exts[extension[0]]
-            _ext1 = self.exts[extension[1]]
-            appended_ext = _ext0 + _ext1
-            return appended_ext
-
-    def get_prog(self, _ext):
-        if not _ext:
-            print("Invalid extension")
-            return None
-        _prog = self.progs[_ext]
-        if not _prog:
-            print("We don't support this file type")
+def _add_config(extension, encoding, prog, attrs):
+    logger.logd('add_config called:', extension, encoding, prog, attrs)
+
+    json_data = None
+    with open(config.EXTRACTOR_CONFIG_FILE, 'r') as f:
+        try:
+            json_data = json.load(f)
+        except JSONDecodeError:
+            logger.logd('add_config exception')
+            print('Could not add a new configuration to your file')
             return None
-        return _prog
 
-class Extractor:
+    item = {
+        _ext_str: extension,
+        _encoding_str: encoding,
+        _prog_str: prog,
+        _attrs_str: attrs
+    }
 
-    def get_prog(self, extensions):
-        if not extensions:
-            return None
+    json_data[_configs_str].append(item)
+
+    with open(config.EXTRACTOR_CONFIG_FILE, 'w') as outfile:
+        json.dump(json_data, outfile, indent=2)
+    return item
+
+def _configure_new_item(extension, encoding):
+    prog = input('What\'s the program? (No options) ') or None
+    opts = input('Options? ') or None
+    return _add_config(extension, encoding, prog, opts) if prog else None
+
+def _guess_type(filename):
+    return mimetypes.guess_type(filename, False)
+
+def extract(filename):
```

Comment:
maybe we should move extract up there, as it is the public method. it will be immediatelly visible when opening the file